src/NavbarOffcanvas.tsx

```diff
@@ -1,13 +1,80 @@
 import * as React from 'react';
 import { useContext } from 'react';
+import useBreakpoint from '@restart/hooks/useBreakpoint';
+import classNames from 'classnames';
+import { useBootstrapPrefix } from './ThemeProvider';
 import Offcanvas, { OffcanvasProps } from './Offcanvas';
 import NavbarContext from './NavbarContext';
 
-const NavbarOffcanvas = React.forwardRef<HTMLDivElement, OffcanvasProps>(
-  (props, ref) => {
+export type NavbarOffcanvasProps = Omit<OffcanvasProps, 'show'>;
+
+const NavbarOffcanvas = React.forwardRef<HTMLDivElement, NavbarOffcanvasProps>(
+  (
+    {
+      className,
+      bsPrefix,
+      backdrop,
+      backdropClassName,
+      keyboard,
+      scroll,
+      placement,
+      autoFocus,
+      enforceFocus,
+      restoreFocus,
+      restoreFocusOptions,
+      onShow,
+      onHide,
+      onEscapeKeyDown,
+      onEnter,
+      onEntering,
+      onEntered,
+      onExit,
+      onExiting,
+      onExited,
+      ...props
+    },
+    ref,
+  ) => {
     const context = useContext(NavbarContext);
+    bsPrefix = useBootstrapPrefix(bsPrefix, 'offcanvas');
+    const hasExpandProp = typeof context?.expand === 'string';
+    const shouldExpand = useBreakpoint(
+      (hasExpandProp ? context.expand : 'xs') as any,
+      'up',
+    );
 
-    return <Offcanvas ref={ref} show={!!context?.expanded} {...props} />;
+    return hasExpandProp && shouldExpand ? (
+      <div
+        ref={ref}
+        {...props}
+        className={classNames(className, bsPrefix, `${bsPrefix}-${placement}`)}
+      />
+    ) : (
+      <Offcanvas
+        ref={ref}
+        show={!!context?.expanded}
+        bsPrefix={bsPrefix}
+        backdrop={backdrop}
+        backdropClassName={backdropClassName}
+        keyboard={keyboard}
+        scroll={scroll}
+        placement={placement}
+        autoFocus={autoFocus}
+        enforceFocus={enforceFocus}
+        restoreFocus={restoreFocus}
+        restoreFocusOptions={restoreFocusOptions}
+        onShow={onShow}
+        onHide={onHide}
+        onEscapeKeyDown={onEscapeKeyDown}
+        onEnter={onEnter}
+        onEntering={onEntering}
+        onEntered={onEntered}
+        onExit={onExit}
+        onExiting={onExiting}
+        onExited={onExited}
```

Comment:
why change this from `{...props}`? They're being forwarded anyway

Reply:
i mean _most_ of these

Reply:
lol yeah it's a bit ugly.  I pulled it out because I also spread the remainder of the props into the div above for the navbar expand case.